mixer/adapter/stackdriver/trace/trace.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 the Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package trace contains a tracespan adapter for Stackdriver trace.
+package trace
+
+import (
+	"context"
+	"fmt"
+
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/plugin/ochttp/propagation/b3"
+	"go.opencensus.io/trace"
+
+	"istio.io/istio/mixer/adapter/stackdriver/config"
+	"istio.io/istio/mixer/adapter/stackdriver/helper"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/tracespan"
+)
+
+type (
+	builder struct {
+		types map[string]*tracespan.Type
+		mg    *helper.MetadataGenerator
+		cfg   *config.Params
+	}
+
+	handler struct {
+		te      trace.Exporter
+		sampler trace.Sampler
+	}
+)
+
+var (
+	// compile-time assertion that we implement the interfaces we promise
+	_ tracespan.HandlerBuilder = &builder{}
+	_ tracespan.Handler        = &handler{}
+)
+
+// NewBuilder returns a builder implementing the tracespan.HandlerBuilder interface.
+func NewBuilder(mg *helper.MetadataGenerator) tracespan.HandlerBuilder {
+	return &builder{mg: mg}
+}
+
+func (b *builder) SetTraceSpanTypes(types map[string]*tracespan.Type) {
+	b.types = types
+}
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.cfg = cfg.(*config.Params)
+}
+func (b *builder) Validate() *adapter.ConfigErrors {
+	return nil
+}
+
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+	exporter, err := getExporterFunc(ctx, env, b.cfg)
```

Comment:
The metadata generator could help fill project id if it is not set in adapter config, so that a generalized config could be used for different projects as long as it is running gce, but you can leave this to me, I will add this later.